src/engine/source/indexerconnector/test/unit/serverSelector_test.cpp

```diff
@@ -162,21 +180,59 @@ TEST_F(ServerSelectorTest, TestGetNextBeforeAndAfterHealthCheck)
  */
 TEST_F(ServerSelectorTest, TestGextNextWhenThereAreNoAvailableServers)
 {
-    const auto hostRedServer {m_servers.at(1)};
 
     m_servers.clear();
-    m_servers.emplace_back(hostRedServer);
+    m_servers.emplace_back(RED_SERVER);
+
+    // Set up the expectations for the MockHTTPRequest
+    EXPECT_CALL(*spHTTPRequest, get(::testing::_, ::testing::_, ::testing::_))
+        .WillRepeatedly(::testing::Invoke(
+            [](RequestParameters requestParameters,
+               PostRequestParameters postRequestParameters,
+               ConfigurationParameters /*configurationParameters*/)
+            {
+                const auto& url = requestParameters.url.url(); // Convert URL to string if needed
+                std::string response;
+
+                if (url == GREEN_SERVER + "/_cat/health?v")
+                {
+                    // Simulate a successful response for the green server
+                    response = "epoch\ttimestamp\tcluster\tstatus\tnode.total\tnode.data\tdiscovered_cluster_"
+                               "manager\tshards\tpri\trelo\tinit\tunassign\tpending_tasks\tmax_task_wait_time\tactive_"
+                               "shards_percent\n1725296432\t17:00:32\twazuh-"
+                               "cluster\tgreen\t1\t1\ttrue\t47\t47\t0\t0\t0\t0\t-\t100.0%\n";
+                    postRequestParameters.onSuccess(response.c_str());
+                }
+                else if (url == RED_SERVER + "/_cat/health?v")
+                {
+                    // Simulate a failed response for the red server
+                    response = "epoch\ttimestamp\tcluster\tstatus\tnode.total\tnode.data\tdiscovered_cluster_"
+                               "manager\tshards\tpri\trelo\tinit\tunassign\tpending_tasks\tmax_task_wait_time\tactive_"
+                               "shards_percent\n1725296432\t17:00:32\twazuh-"
+                               "cluster\tred\t1\t1\ttrue\t47\t47\t0\t0\t0\t0\t-\t100.0%\n";
+                    postRequestParameters.onError(response.c_str(), 200);
+                }
+                else
+                {
+                    // Unknown server, simulate an error
+                    postRequestParameters.onError("Unknown server", 404);
+                    return;
```

Comment:
Improve: Missing yellow server

Reply:
How do we handle the yellow cluster? The previous tests only considered the green or red state. I think the logic is equal to the red server

Reply:
Yes, logic is equal to red server

Reply:
Great, I will add a case for yellow server :)

Reply:
Done!